Foxtrot/Driver/Rewriting/Utils/TypeNodeExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Compiler;
+using System.Diagnostics.Contracts;
+
+namespace Microsoft.Contracts.Foxtrot.Utils
+{
+    internal static class TypeNodeExtensions
+    {
+        public static int TemplateArgumentsCount(this TypeNode typeNode)
+        {
+            Contract.Requires(typeNode != null);
+
+            return typeNode.TemplateArguments == null ? 0 : typeNode.TemplateArguments.Count;
+        }
+
+        public static int CountOrDefault(this TypeNodeList list)
```

Comment:
:arrow_right: Let's create an issue for it. Initially mark it as **blocked** on updating to development in VS 2015, but also mark it with an **easy** label to draw in new contributors.